### PR TITLE
Reconfigure systemd services

### PR DESCRIPTION
**Pull request by peeech**
This PR is aiming to set correct order and properties of hosting-related systemd services on HPOS.

Required steps:
- [x] update systemd dependency tree
- [x] update services properties
- [x] ~~create `lair-shim` service?~~
- [x] analyze tests from perspective on new order

### CODE REVIEW

File: overlays/holo-nixpkgs/holochain/versions.nix [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   hpos = {
-    rev = "358018b1ae3c413bd2b58c0b12752c7b38779c09";
-    sha256 = "1af2vchilwdky6fjxvn93gr6ki793kwczry8nxm74qcikkx6rwb7";
-    cargoSha256 = "1nzr98dq8vg4av7p6ff19922k0mlpzcdzvrhwvdvbywawvawadx7";
+    rev = "12dd1d51dfae597b1f89ca1bbf57c475ff971427";
```
Comment (zo-el) [inline overlays/holo-nixpkgs/holochain/versions.nix:3]: What is this pointing to?

File: modules/services/configure-holochain.nix [hunk 2]
```diff
@@ -26,8 +26,8 @@ in
 
   config = mkIf (cfg.enable) {
     systemd.services.configure-holochain = {
-      after = [ "network.target" "holochain.service" "hpos-holochain-api.service" ];
```
Comment (zo-el) [inline modules/services/configure-holochain.nix:29]: What is going to trigger `configure-holochain` to start up after `holochain.service`?

File: overlays/holo-nixpkgs/holochain/versions.nix (hunk 1 above)
Comment (peeech) [reply overlays/holo-nixpkgs/holochain/versions.nix:3]: To [this](https://github.com/holochain/holochain/pull/739) PR

File: modules/services/configure-holochain.nix (hunk 2 above)
Comment (peeech) [reply modules/services/configure-holochain.nix:29]: Ready state of `hpos-holochain-api.service` unit.

**Comment by peeech** [comment]
Unfortunately `develop` of holochain is not compiling on HPOS because of an error in using defaut features. I need to fix it first :-(

File: modules/services/configure-holochain.nix (hunk 2 above)
Comment (zo-el) [reply modules/services/configure-holochain.nix:29]: Perfect

**Comment by peeech** [comment]
> Unfortunately `develop` of holochain is not compiling on HPOS because of an error in using defaut features. I need to fix it first :-(

Fixed in https://github.com/holochain/holochain/pull/742

**Comment by alastairong** [comment]
Looks great. I don't think I have ever seen a failure happen because Holochain service wasn't ready, but it's an important resiliency upgrade nonetheless.

**Comment by peeech** [comment]
Merging this PR requires deciding which `holochain` revision are we gonna use

**Comment by peeech** [comment]
@alastairong this one needs your approval as a code owner

**Comment by alastairong** [comment]
I love that we have a second set of tests.

**Review by alastairong — APPROVED** [review]
Nice and tidy!

**Pull request merged**